File: python/services/openai_service.py
import openai

class OpenAIService:
    def __init__(self, api_key: str, model: str):
        self.api_key = api_key
        self.model = model
        openai.api_key = self.api_key

    def analyze_dsa(self, prompt: str):
        response = openai.ChatCompletion.create(
            model=self.model,
            messages=[{"role": "user", "content": prompt}]
        )
        return response.choices[0].message['content']

File: nodejs/services/openaiService.js
const openai = require('openai');

class OpenAIService {
    constructor(apiKey, model) {
        this.apiKey = apiKey;
        this.model = model;
        this.client = new openai.OpenAI(apiKey);
    }

    async analyzeDsa(prompt) {
        try {
            const response = await this.client.chatCompletions.create({
                model: this.model,
                messages: [{ role: 'user', content: prompt }]
            });
            return response.choices[0].message.content;
        } catch (error) {
            throw new Error('Failed to analyze DSA question: ' + error.message);
        }
    }
}

module.exports = OpenAIService;
